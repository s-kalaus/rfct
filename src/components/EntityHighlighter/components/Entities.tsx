import React from 'react';
import { IEntity } from '~/interfaces/entity';
import { Entity } from './Entity';

export interface IProps {
  text: string;
  entities: IEntity[];
}

export const Entities = ({ text, entities }: IProps) => (
  <>
    {entities.map((entity) => (
      <Entity text={text} entity={entity} key={JSON.stringify(entity)} />
    ))}
  </>
);
