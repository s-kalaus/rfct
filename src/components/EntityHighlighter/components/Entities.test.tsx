import React from 'react';
import { render } from '@testing-library/react';
import { Entities } from './Entities';

describe('Entities', () => {
  const text = 'Text';
  const entities = [{ selectionStart: 0, selectionEnd: 0, label: 'Label' }];

  test('renders component', () => {
    const { container } = render(<Entities text={text} entities={entities} />);
    expect(container).toMatchSnapshot();
  });
});
