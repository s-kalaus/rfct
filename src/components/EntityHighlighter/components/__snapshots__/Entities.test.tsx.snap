// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Entities renders component 1`] = `
<div>
  <div
    style="text-align: left; position: absolute; top: 6px; left: 6px; right: 6px; color: transparent; pointer-events: none; padding: 0px; white-space: pre-wrap; word-break: break-word; font-family: source-code-pro, Menlo, Monaco, Consolas, \\"Courier New\\", monospace; font-size: 14px;"
  >
    
    <span
      style="opacity: 0.3; background-color: rgb(1, 255, 112);"
    />
    Text
  </div>
</div>
`;
